src/Language/PureScript/Errors.hs

```diff
@@ -636,6 +638,15 @@ prettyPrintSingleError (PPEOptions codeColor full level showWiki) e = flip evalS
         where
         go TUnknown{} = True
         go _ = False
+    renderSimpleErrorMessage (AmbiguousTypeVariables t (Constraint nm ts _)) =
+      paras [ line "Inferred type"
+            , markCodeBox $ typeAtomAsBox t
+            , line " has constraints, but we don't know what type values they require. Please add an explicit type declaration for: "
```

Comment:
Sorry to be picky, but can this be just "The inferred type ... has type variables which are not mentioned in the body of the type. Consider adding a type annotation"? Thanks.